BUILD.cmake.md

```diff
@@ -0,0 +1,203 @@
+
+_NOTE_: HIDAPI packages built by CMake can be used with `pkg-config`, as if built with [Autotools](BUILD.autotools.md).
+
+### MSVC and Ninja
+It is possible to build a CMake project (including HIDAPI) using MSVC compiler and Ninja (for medium and larger projects it is so much faster than msbuild).
+
+For that:
+1) Open cmd.exe;
+2) Setup MSVC build environment variables, e.g.: `vcvarsall.bat x64`, where:
+	- `vcvarsall.bat` is an environment setup script of your MSVC toolchain installation;<br/>For MSVC 2019 Community edition it is located at: `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\`;
+	- `x64` -a target architecture to build;
+3) Follow general build steps, and use `Ninja` as a generator.
+
+### Using HIDAPI in a CMake project
+
+When HIDAPI is used as a standalone package (either installed into the system or built manually and installed elsewhere), the simplest way to use it is as showed in the example:
+
+```cmake
+project(my_application)
+
+add_executable(my_application main.c)
+
+find_package(hidapi REQUIRED)
+target_link_libraries(my_application PRIVATE hidapi::hidapi)
+```
+
+If HIDAPI isn't installed in your system, or `find_package` cannot find HIDAPI by default for any other reasons,
+the recommended way manually specify which HIDAPI package to use is via `hidapi_ROOT` CMake variable, e.g.:
+`-Dhidapi_ROOT=<path to HIDAPI installation prefix>`.
+
+_NOTE_: usage of `hidapi_ROOT` is only possible (and recommended) with CMake 3.12 and higher. For older versions of CMake you'd need to specify [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html#variable:CMAKE_PREFIX_PATH) instead.
+
+Check with [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html) documentation if you need more details.
+
+Available CMake targets after successful `find_package(hidapi)`:
+- `hidapi::hidapi` - indented to be used in most cases;
+- `hidapi::include` - if you need only to include `<hidapi.h>` but not link against the library;
+- `hidapi::winapi` - same as `hidapi::hidapi` on Windows; available only on Windows;
+- `hidapi::darwin` - same as `hidapi::hidapi` on macOS; available only on macOS;
+- `hidapi::libusb` - available when libusb backend is used/available;
+- `hidapi::hidraw` - available when hidraw backend is used/available on Linux;
+
+**NOTE**: on Linux often both `hidapi::libusb` and `hidapi::hidraw` backends are available; in that case `hidapi::hidapi` is an alias for **`hidapi::libusb`**. The motivation is that hidraw backend doesn't have an implementation of all of the HIDAPI functions due to lack of Linux kernel support. If libusb backend isn't built at all (`hidapi::hidraw` is the only target) - `hidapi::hidapi` is an alias for `hidapi::hidraw`.
```

Comment:
What is not implemented by the hidraw backend? Where is that documented? I am doubting libusb should be the default backend because it can't work with Bluetooth HID devices.

Reply:
> What is not implemented by the hidraw backend?

https://github.com/libusb/hidapi/blob/7620bc7faac7613ff8ea9efbbd54b1a0d719cf76/linux/hid.c#L1069
https://github.com/libusb/hidapi/blob/7620bc7faac7613ff8ea9efbbd54b1a0d719cf76/linux/hid.c#L1108

> I am doubting libusb should be the default backend 

I had a hard time making that decision too. I guess I'm still open for discussion on this matter.

Reply:
Is it still true that the hidraw API does have the features needed for hidapi to implement these APIs for the hidraw backend? Maybe they have been implemented since the hidraw backend was written.

Reply:
#259 is one issue tracker to implement hid_get_input_report() for new kernel versions. Not so sure about hid_get_indexed_string().

Reply:
Still I will vote to set hidraw backend as the default for Linux as it is the way to go for most use cases down the road.

Reply:
updated